catboost/python-package/ut/medium/test.py

```diff
@@ -2050,6 +2051,52 @@ def test_get_metadata_notrain():
         assert str(i + 1) in model.get_metadata()
 
 
+# tests for single objects
+
+def get_single_object_from_dataset(pool):
+    test_list = open(data_file(pool, 'test_small')).read().split('\n')[:-1]
+    for i in range(len(test_list)):
+        test_list[i] = test_list[i].split('\t')
+        for j in range(len(test_list[i])):
+            try:
+                test_list[i][j] = float(test_list[i][j])
+            except:
+                test_list[i][j] = test_list[i][j]
+    return random.choice(test_list)
+
+
+@pytest.mark.parametrize('pool', ['higgs', 'adult'])
+def test_single_object_features():
+    train_pool = Pool(data_file(pool, 'train_small'), column_description=data_file(pool, 'train.cd'))
+    test_object = get_single_object_from_dataset(pool)
+    model = CatBoost({'iterations': 20, 'random_seed': 0, 'loss_function': 'RMSE', 'task_type': task_type, 'devices': '0'})
+    model.fit(train_pool)
+    assert model.predict(test_object) == model.predict([test_object])[0]
+    assert list(model.staged_predict(test_object)) == list(list(model.staged_predict([test_object]))[0])
+
+
+@pytest.mark.parametrize('pool', 'adult')
+def test_single_object_regressor():
+    train_pool = Pool(data_file(pool, 'train_small'), column_description=data_file(pool, 'train.cd'))
+    model = CatBoostRegressor(iterations=2, learning_rate=0.03, random_seed=0, task_type=task_type, devices='0')
+    model.fit(train_pool)
+    test_object = get_single_object_from_dataset(pool)
+    assert model.predict(test_object) == model.predict([test_object])[0]
+    assert list(model.staged_predict(test_object)) == list(list(model.staged_predict([test_object]))[0])
+
+
+@pytest.mark.parametrize('pool', 'adult')
```

Comment:
are you sure this compiles? I think you must provide `pool` as a parameter for the test, and you probably need to parametrize it as 
```suggestion
 @pytest.mark.parametrize('pool', ['adult'])
```